iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOManager.java

```diff
@@ -48,4 +54,21 @@ public FileIO getFileIO(StorageType.Type storageType) throws IllegalArgumentExce
       throw new IllegalArgumentException("FileIO not supported for storage type: " + storageType);
     }
   }
+
+  /**
+   * Returns the Storage implementation for the given FileIO.
+   *
+   * @param fileIO, the FileIO for which the Storage implementation is required
+   * @return Storage implementation for the given FileIO
+   * @throws IllegalArgumentException if the FileIO is not configured
+   */
+  public Storage getStorage(FileIO fileIO) {
+    if (fileIO.equals(hdfsFileIO)) {
+      return hdfsStorage;
+    } else if (fileIO.equals(localFileIO)) {
+      return localStorage;
```

Comment:
isn't this better to use `instanceOf` ? 
Not a fan of ^ either (result of this method sometimes vary between jvm versions, there should be a lot of SO posts about this) -- but having a object equality check sounds even more strange ... Like, did we ensure equal method is implemented in our favor?

Reply:
I prefer equals because the exact same object is being compared, not a copy, not a child. This comparison is most restrictive and safe. Let's keep the same,  until we come across use cases where less restrictive comparison is required?